src/btree/bt_curnext.c

```diff
@@ -423,7 +423,8 @@ __cursor_row_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
 restart_read_page:
         rip = &page->pg_row[cbt->slot];
         WT_RET(__cursor_row_slot_key_return(cbt, rip, &kpack, &kpack_used));
-        WT_RET(__wt_txn_read(session, cbt, WT_ROW_UPDATE(page, rip), NULL, &upd));
+        WT_RET(__wt_txn_read(
+          session, cbt, &cbt->iface.key, WT_RECNO_OOB, WT_ROW_UPDATE(page, rip), NULL, &upd));
```

Comment:
When do we use `cbt->iface.key` vs `cbt->tmp`?

Reply:
`WT_CURSOR_BTREE.iface.key` is the same as `WT_CURSOR.key`, that is, it's the application cursor key. So, it's only used when the app is setting a key, or a key is being returned to the app.

`WT_CURSOR_BTREE.tmp` is used for a few different things unfortunately, this change greatly expands how often it's used. In this particular case, it's set by the row-store search routine. The reason is the row-store search routine may have done a bunch of work to create the key (for example, prefix compression pages), and so we save off a copy so it's faster when we return that key to the application. We can't use `WT_CURSOR.key` for that purpose because if we have to restart the search for some reason, we might need the original key as set by the app.